Source/Common/NMR_StringUtils.cpp

```diff
@@ -161,6 +161,10 @@ namespace NMR {
 		{
 			throw CNMRException(NMR_ERROR_EMPTYSTRINGTODOUBLECONVERSION);
 		}
+		else if (answer.ptr && answer.ptr[0] == ',') // Invalidate comma as decimal separator
+		{
+			throw CNMRException(NMR_ERROR_INVALIDSTRINGTODOUBLECONVERSION);
```

Comment:
let's also add a test for that, please.

Reply:
I would prefer that we do never add such style of code. While it is shorter, it is a landmine and who knows on what situation and platform (or little change by somebody) answer.ptr[0] is evaluated when answer.ptr == nullptr.

Making two if statements out of it is much easier to read and much safer.

Reply:
if(answer.ptr){
    if(answer.ptr[0] == ','){
        throw here
    }
}

I can make this into two if statements for readability, but that would mean the same. Did you mean anything else?

Reply:
That's what I meant, yes :)